exercises/practice/affine-cipher/.meta/solutions/lib/AffineCipher.pm

```diff
@@ -0,0 +1,47 @@
+package AffineCipher;
+
+use strict;
+use warnings;
+use experimental qw<signatures postderef postderef_qq>;
+
+use Exporter qw<import>;
+our @EXPORT_OK = qw<encode decode>;
+
+sub encode ( $phrase, $a, $b ) {
+    die "a and m must be coprime." unless _coprime($a);
+    my $ret = '';
+    foreach ( split( //, $phrase ) ) {
+        next        if $_ eq ' ';
+        $ret .= ' ' if ( $_ =~ /[[:alpha:]\d]/ && !( ( length($ret) + 1 ) % 6 ) );
+        if ( ord( lc($_) ) < 97 ) {
+            $ret .= $_ if ( $_ =~ /\d/ );
+        } else {
+            $ret .= chr( ( ( $a * ( ord( lc($_) ) - 97 ) + $b ) % 26 ) + 97 );
+        }
+    }
+    return $ret;
```

Comment:
In both cases of `encode` and `decode` the return appears to b the message (regardless of the  type of message).

I would not use `return $ret` as it does not describe "message" as the return, in either case.